src/compiler/abstract_compiler.nit

```diff
@@ -2689,10 +2722,15 @@ redef class AMethPropdef
 		var pname = mpropdef.mproperty.name
 		var cname = mpropdef.mclassdef.mclass.name
 		var ret = mpropdef.msignature.return_mtype
-		if ret != null then
+                var compiler = v.compiler
+                # WARNING: we must not resolve the return type when it's a functional type.
+                # Otherwise, we get a compile error exactly here. This weird behavior doesn't affect
+                # the inner mecanics of callref since the return type is already solved by
+                # `routine_ref_call`
+                if ret != null and not compiler.all_routine_types_name.has(cname) then
```

Comment:
I couldn't reproduce the weird behavior. Do you have a minimal example? (also put it in the `/tests` directory)

Reply:
I added `tests/test_callref.nit`, this behavior only occured if you compile in semi-global.